gnucash/register/ledger-core/split-register-control.cpp

```diff
@@ -668,23 +669,14 @@ static Split *
 gnc_find_split_in_account_by_memo (Account *account, const char *memo,
                                    gboolean unit_price)
 {
-    if (account == NULL) return NULL;
+    if (account == nullptr) return nullptr;
 
-    Split *rv = nullptr;
-    auto splits = xaccAccountGetSplitList (account);
-    for (auto slp = g_list_last (splits); !rv && slp; slp = slp->prev)
-    {
-        auto split = GNC_SPLIT(slp->data);
-        Transaction *trans = xaccSplitGetParent (split);
-
-        split = gnc_find_split_in_trans_by_memo (trans, memo, unit_price);
-
-        if (split)
-            rv = split;
-    }
+    const auto& splits = xaccAccountGetSplits (account);
+    for (auto it = splits.rbegin(); it != splits.rend(); it++)
+        if (auto split = gnc_find_split_in_trans_by_memo (xaccSplitGetParent (*it), memo, unit_price))
+            return split;
```

Comment:
Surely you can do better than a raw loop here.

Reply:
find_if will search correctly; but it returns the iterator rather than the result of the predicate. hence the above construct to avoid calling `gnc_find_split_in_trans_by_memo` again

Reply:
Mrmmph. Transaction split lists tend to be very short so it's not a big deal to loop over one twice, but this is sufficiently simple that I guess there's not much clarity gained with an algo.